src/maker/server.rs

```diff
@@ -224,28 +224,66 @@ fn setup_fidelity_bond(maker: &Arc<Maker>, maker_address: &str) -> Result<(), Ma
                 .get_wallet()
                 .write()?
                 .create_fidelity(amount, locktime);
+
             match fidelity_result {
                 // Wait for sufficient fund to create fidelity bond.
                 // Hard error if fidelity still can't be created.
                 Err(e) => {
+
                     if let WalletError::InsufficientFund {
                         available,
                         required,
                     } = e
                     {
-                        log::warn!("Insufficient fund to create fidelity bond.");
-                        let amount = required - available;
-                        let (_, addr, _) = maker
-                            .get_wallet()
-                            .read()?
-                            .get_next_fidelity_address(locktime)?;
-                        log::info!("Send {} sats to {}", amount, addr);
-                        if cfg!(feature = "integration-test") {
-                            sleep(Duration::from_secs(3));
-                        } else {
-                            sleep(Duration::from_secs(300)); // Wait for 5 mins in production
+                        
+                        loop {
+
+                            let (_, addr, _) = maker
+                                .get_wallet()
+                                .read()?
+                                .get_next_fidelity_address(locktime)?;
```

Comment:
Ah this is the bug. 

`get_next_fidelity_address` gives an address which is not included in the `Wallet::descriptors_to_import()` in the import list. Thus it's not getting caught in the sync.

Instead this should be a `get_external_address` call. The wallet first gets fund in the external address. Then it creates the fidelity transaction spending from that address. 

Corollary: `get_next_fidelity_address` should be `pub(super)` as it should not be a user facing API. Only the wallet should use it in `create_fidelity()` internally.

It's my bad. I should have seen this before while writing it.